### PR TITLE
introduce --no-build option

### PR DESCRIPTION
**Pull request by ndeloof**
**What I did**
Introduced --no-build option to disable building images from source

**Related issue**
https://github.com/docker/compose-cli/issues/1291

**(not mandatory) A picture of a cute animal, if possible in relation with what you did**

### CODE REVIEW

File: cli/cmd/compose/up.go [hunk 1]
```diff
@@ -284,10 +289,18 @@ func setup(ctx context.Context, opts composeOptions, services []string) (*client
 		project.Services[0].DomainName = opts.DomainName
 	}
 	if opts.Build {
-		for _, service := range project.Services {
+		for i, service := range project.Services {
 			service.PullPolicy = types.PullPolicyBuild
+			project.Services[i] = service
```
Comment (gtardif) [inline cli/cmd/compose/up.go:294]: did we have a bug here not reaffecting modified service ?

File: cli/cmd/compose/create.go [hunk 2]
```diff
@@ -34,10 +35,14 @@ func createCommand(p *projectOptions) *cobra.Command {
 		Use:   "create [SERVICE...]",
 		Short: "Creates containers for a service.",
 		RunE: func(cmd *cobra.Command, args []string) error {
+			if opts.Build && opts.noBuild {
+				return fmt.Errorf("--build and --no-build are incompatible")
+			}
```
Comment (gtardif) [inline cli/cmd/compose/create.go:40]: Nit: we should also add a check for force-recreate and no-recreate, next to this one. (Check already exists in the up command)

**Review by gtardif — APPROVED** [review]
LGTM, small improvement on missing check for force-recreate / no-recreate flag next to no-build

File: cli/cmd/compose/up.go (hunk 1 above)
Comment (ndeloof) [reply cli/cmd/compose/up.go:294]: seems so

**Pull request merged**